mmv1/third_party/terraform/fwprovider/framework_provider_project_test.go

```diff
@@ -0,0 +1,257 @@
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+)
+
+func TestAccFwProvider_project(t *testing.T) {
+	testCases := map[string]func(t *testing.T){
+		"config takes precedence over environment variables":                           testAccFwProvider_project_configPrecedenceOverEnvironmentVariables,
+		"when project is unset in the config, environment variables are used":          testAccFwProvider_project_precedenceOrderEnvironmentVariables,
+		"when project is set to an empty string in the config the value isn't ignored": testAccFwProvider_project_emptyStringValidation,
+		"when project is unknown in the config, environment variables are used":        testAccFwProvider_project_unknownHandling,
+	}
+
+	for name, tc := range testCases {
+		// shadow the tc variable into scope so that when
+		// the loop continues, if t.Run hasn't executed tc(t)
+		// yet, we don't have a race condition
+		// see https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			tc(t)
+		})
+	}
+}
+
+func testAccFwProvider_project_configPrecedenceOverEnvironmentVariables(t *testing.T) {
+	acctest.SkipIfVcr(t) // Test doesn't interact with API
+
+	project := envvar.GetTestProjectFromEnv()
+
+	// set all possible project env vars to other value; show they aren't used
+	for _, v := range envvar.ProjectEnvVars {
+		t.Setenv(v, "foobar")
+	}
+
+	context := map[string]interface{}{
+		"project": project,
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		// No PreCheck for checking ENVs
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccFwProvider_projectInProviderBlock(context),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.google_provider_config_plugin_framework.default", "project", project),
+				),
+			},
+		},
+	})
+}
+
+func testAccFwProvider_project_precedenceOrderEnvironmentVariables(t *testing.T) {
+	acctest.SkipIfVcr(t) // Test doesn't interact with API
+
+	/*
+		These are all the ENVs for project, and they are in order of precedence.
+		GOOGLE_PROJECT
+		GOOGLE_CLOUD_PROJECT
+		GCLOUD_PROJECT
+		CLOUDSDK_CORE_PROJECT
+	*/
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				// Unset all ENVs for project
+				PreConfig: func() {
+					for _, v := range envvar.ProjectEnvVars {
+						t.Setenv(v, "")
+					}
+				},
+				Config: testAccFwProvider_projectInEnvsOnly(),
+				Check: resource.ComposeTestCheckFunc(
+					// Differing behavior between SDK and PF; the attribute is NOT found here.
+					// This reflects the different type systems used in the SDKv2 and the plugin-framework
```

Comment:
Differing behaviour  = In PF tests the field is missing when there is no value for it, but in the SDK the field is set to an empty string.